channel.go

```diff
@@ -132,7 +133,38 @@ func (c *vSockChannel) setup() error {
 }
 
 func (c *vSockChannel) wait() error {
-	return nil
+	if vsockHostPort == uint32(0) {
+		return fmt.Errorf("fail to get vsock host port %d", vsockHostPort)
+	}
+
+	cancel := make(chan bool)
+	conn := make(chan bool)
+	go func() {
+		for {
+			select {
+			case <-cancel:
+				return
+			default:
+			}
+
+			// Let the agent launch a new connection to vsock vhost port
+			// to tell it the vsock transport in guest is ready
+			c, err := vsock.Dial(unix.VMADDR_CID_HOST, vsockHostPort)
+			if err == nil {
+				conn <- true
+				c.Close()
+				return
+			}
+		}
+	}()
+
+	select {
+	case <-conn:
+		return nil
+	case <-time.After(vsockReadyTimeout):
+		cancel <- true
```

Comment:
Does this cancelation really work?  When time.After(vsockReadyTimeout) expires we send true to the goroutine that is blocked in vsock.Dial().  Nothing interrupts vsock.Dial() so I guess it will continue waiting until the vsock connection itself completes, failsm or times out.

Goroutines probably aren't necessary here: you can just call vsock.Dial().  It will time out after 2 seconds or you can use setsockopt(SO_VM_SOCKETS_CONNECT_TIMEOUT) to set a different timeout.